tradeexecutor/ethereum/execution.py

```diff
@@ -5,34 +5,435 @@
 from collections import Counter
 from decimal import Decimal
 from typing import List, Dict, Set, Tuple
+from abc import ABC, abstractmethod
 
 from eth_account.datastructures import SignedTransaction
 from eth_typing import HexAddress
 from hexbytes import HexBytes
 from web3 import Web3
-from web3.contract import Contract
 
 from eth_defi.abi import get_deployed_contract
 from eth_defi.gas import GasPriceSuggestion, apply_gas, estimate_gas_fees
 from eth_defi.hotwallet import HotWallet
-from eth_defi.revert_reason import fetch_transaction_revert_reason
 from eth_defi.token import fetch_erc20_details, TokenDetails
 from eth_defi.confirmation import wait_transactions_to_complete, \
     broadcast_and_wait_transactions_to_complete, broadcast_transactions
-from eth_defi.uniswap_v2.deployment import UniswapV2Deployment, FOREVER_DEADLINE,  mock_partial_deployment_for_analysis
-from eth_defi.uniswap_v2.fees import estimate_sell_price_decimals
-from eth_defi.uniswap_v2.analysis import analyse_trade_by_hash, TradeSuccess, analyse_trade_by_receipt
+from eth_defi.uniswap_v2.deployment import UniswapV2Deployment, FOREVER_DEADLINE 
+from eth_defi.trade import TradeSuccess, TradeFail
+from eth_defi.revert_reason import fetch_transaction_revert_reason
+
 from tradeexecutor.state.state import State
-from tradeexecutor.state.trade import TradeExecution
+from tradeexecutor.state.trade import TradeExecution, TradeStatus
 from tradeexecutor.state.blockhain_transaction import BlockchainTransaction
-from tradeexecutor.state.identifier import AssetIdentifier, TradingPairIdentifier
+from tradeexecutor.state.freeze import freeze_position_on_failed_trade
+from tradeexecutor.state.identifier import AssetIdentifier
+from tradeexecutor.ethereum.uniswap_v2_routing import UniswapV2SimpleRoutingModel, UniswapV2RoutingState
+from tradeexecutor.ethereum.uniswap_v3_routing import UniswapV3SimpleRoutingModel, UniswapV3RoutingState
+from tradeexecutor.strategy.execution_model import ExecutionModel
 
 logger = logging.getLogger(__name__)
 
+routing_models = UniswapV2SimpleRoutingModel | UniswapV3SimpleRoutingModel # TODO create enum, actually just use ehtereum Base
+routing_states = UniswapV2RoutingState | UniswapV3RoutingState # TODO create enum, actually just use ethereum base
+
 
 class TradeExecutionFailed(Exception):
     """Our Uniswap trade reverted"""
 
+# TODO check with tradeexuctor.strategy.execution ExecutionModel
```

Comment:
Please do not move existing module functions to a class without discussing beforehand why.

I do not see a reason here why it was done and it is not needed.